File: examples/perl/ccu_v3.pl
#! /usr/bin/env perl
# Very basic script demonstrating diagnostic tools functionality
#
# Copyright 2015 Akamai Technologies, Inc. All Rights Reserved.
# 
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
#
# You may obtain a copy of the License at 
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#

use Akamai::Edgegrid;
use Config::IniFiles;
use JSON;
use strict;
use warnings;
use URI; 
use Getopt::Long;

my %args;

my $debug = 0;
my $verbose = 0;

GetOptions ("debug" => \$debug, 
            "verbose"  => \$verbose)
  or die("Error in command line arguments\n");

my $config_file = "$ENV{HOME}/.edgerc";
my $section = "ccu";

my $agent = new Akamai::Edgegrid( 
			config_file => $config_file, 
			section   => $section,
			debug     => $debug );
my $baseurl = "https://" . $agent->{host};
my $json = JSON->new->utf8->pretty(1);

#######################
# MAIN                #
#######################

########################
# Post a purge request #
########################

my $req = HTTP::Request->new(
	POST => "$baseurl/ccu/v3/invalidate/url");
$req->content_type('application/json');

my $content = '{ "hostname":"bc.akamaiapibootcamp.com", "objects" : [ "/index.html" ] }';
print $content;
$req->content($content);

my $response = $agent->request( $req );
my $purge_object = decode_json($response->content);
if ($verbose == 1) {print $json->encode($purge_object); }

File: examples/node/src/ccu-v3.js
#! /usr/bin/env node

/**
Copyright 2015 Akamai Technologies, Inc. All Rights Reserved.

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.

You may obtain a copy of the License at 

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

========================================================================

Example of using the Akamai OPEN CCU V3 (Fast Purge) API

Note: in order for this to work you need to provision credentials
specifically for CCU - you cannot extend existing credentials to add
CCU as it's managed under "CCU" in the API credential system.

Configure->Organization->Manage APIs
Select "CCU APIs"
Create client collections/clients
Add authorization

Put the credentials in ~/.edgerc as demonstrated by api-kickstart/sample_edgerc
*/

path = require('path'),
    os = require('os'),
    prettyJSON = require('prettyjson'),
    logger = require('./logger'),
    argv = require('minimist')(process.argv.slice(2));

// Akamai EdgeGrid signing library
var EdgeGrid = require('edgegrid');

// Optional command-line arguments
var debug = argv.debug ? true : false;
var verbose = argv.verbose ? true : false;
var headers = argv.headers ? argv.headers : {};

// The result data returned by the purge POST request
var purgePostResult;

// The path to the .edgerc file to use for authentication
var edgercPath = path.join(os.homedir(), "/.edgerc");

// The section of the .edgerc file to use for authentication
var sectionName = "ccu";

// Create a new instance of the EdgeGrid signing library
var eg = new EdgeGrid({
    path: edgercPath,
    section: sectionName,
    debug: debug
});

/**
 * Adds item to be invalidated.
 */
function invalidate() {

    purgeObj = {
        "hostname": "bc.akamaiapibootcamp.com",
        "objects": [
            "/index.html"
        ]
    };

    console.log("Adding data to queue: " + JSON.stringify(purgeObj));

    eg.auth({
        path: "/ccu/v3/invalidate/url",
        method: "POST",
        headers: { 'Content-Type': 'application/json' },
        body: purgeObj
    });

    eg.send(function(data, response) {
        data = JSON.parse(data);
        console.log("Response: ", data);
    });
}

invalidate();
